lib/apimatic-core/logger/configurations/api_logging_configuration.rb

```diff
@@ -0,0 +1,112 @@
+module CoreLibrary
+  # Represents options for configuring logging behavior.
+  class LoggingConfiguration
+    attr_accessor :logger, :log_level, :request_logging_config, :response_logging_config, :mask_sensitive_headers
+
+    # Initializes a new instance of LoggingConfiguration.
+    #
+    # @param logger [LoggerInterface] The logger to use for logging messages.
+    # @param log_level [LogLevel] The log level to determine which messages should be logged.
+    # @param request_logging_config [RequestLoggingConfiguration] Options for logging HTTP requests.
+    # @param response_logging_config [ResponseLoggingConfiguration] Options for logging HTTP responses.
+    # @param mask_sensitive_headers [Boolean] Indicates whether sensitive headers should be masked in logged messages.
+    def initialize(
+      logger: ConsoleLogger.new,
+      log_level: ::Logger::INFO,
```

Comment:
Nits, Non-blocker: instead of having a default value as INFO here, what if we make it part of the SdkLogger class as we did in other langs e.g. Java, Python & C#.

```Python
    def get_logging_level(self):
        if self._api_logging_config.log_level is None:
            return logging.INFO

        return self._api_logging_config.log_level
```

Reply:
We decided to keep this in api logging configuration class